src/cli/server/http_1.rs

```diff
@@ -15,18 +16,51 @@ pub async fn start_http_1(
     let addr = sc.addr();
     let make_svc_single_req = make_service_fn(|_conn| {
         let state = Arc::clone(&sc);
+
         async move {
+            let state = state.clone();
             Ok::<_, anyhow::Error>(service_fn(move |req| {
-                handle_request::<GraphQLRequest>(req, state.app_ctx.clone())
+                let state = state.clone();
+                async move {
+                    let state = state.clone();
+                    match state.app_ctx.blueprint.server.cors_params {
+                        Some(ref cors_params) => {
+                            handle_request_with_cors::<GraphQLRequest>(
+                                req,
+                                cors_params,
+                                state.app_ctx.clone(),
+                            )
+                            .await
+                        }
+                        None => handle_request::<GraphQLRequest>(req, state.app_ctx.clone()).await,
+                    }
+                }
             }))
         }
     });
 
     let make_svc_batch_req = make_service_fn(|_conn| {
         let state = Arc::clone(&sc);
+        let _cors = CorsLayer::permissive();
```

Comment:
so, we're not using it actually and the tower crates as well?

Reply:
This PR is actually incomplete, since I started working on apollo integration. I have copied the entire code from tower's `CorsLayer` to handle cors requests so we do not need tower crates now. The only part remaining is to figure out the correct way to take the input in config. I was trying to do it such that most validation is done automatically during the deserialization